sonar-php-plugin/src/main/java/org/sonar/plugins/php/PHPSensor.java

```diff
@@ -153,11 +160,17 @@ private void processTestsAndCoverage(SensorContext context) {
     new CoverageResultImporter(analysisWarningsWrapper).execute(context);
   }
 
+  private static boolean shouldOptimizeAnalysis(SensorContext context) {
```

Comment:
It looks like this method is called from `AnalysisScanner` only, so should be moved there